packages/backend/src/remote/activitypub/models/tag.ts

```diff
@@ -16,3 +16,34 @@ export function extractApHashtagObjects(tags: IObject | IObject[] | null | undef
 	if (tags == null) return [];
 	return toArray(tags).filter(isHashtag);
 }
+
+export function extractQuoteUrl(tags: IObject | IObject[] | null | undefined): string | null {
+	if (tags == null) return null;
+
+	let quotes: ILink[] = toArray(tags)
+		.filter(isLink)
+		.filter(link => link.mediaType === 'application/activity+json');
```

Comment:
Is this step necessary? `mediaType` isn't required for full semantic understanding. It seems adequate to check only for Links, which ideally have the specified `rel` that you check for immediately below this.

Consider the following JSON:

```json
"tag": [
  {
    "type": "Link",
    "href": "URI",
    "name": "RE: URI",
    "rel": "https://misskey-hub.net/ns#_misskey_quote"
  }
]
```

The current code would ignore this even though it is clearly intended as a misskey quote.

Reply:
One other thing worth pointing out is that [on the associated FEP, I proposed](https://codeberg.org/fediverse/fep/pulls/13#issuecomment-601588) using `rel` instead of `mediaType` as a less ambiguous way of detecting when a Link refers to an ActivityStreams object. Something like this:

```json
"tag": [
  {
    "type": "Link",
    "href": "URI",
    "name": "RE: URI",
    "rel": ["https://www.w3.org/ns/activitystreams", "https://misskey-hub.net/ns#_misskey_quote"]
  }
]
```

Given that the FEP is for "Object Links"

Reply:
Logically I would restructure the flow of this function along the following lines:

1) Check if there's a Link with link-rel of _misskey_quote. If yes, stop.
2) Check if there are any Links with link-rel of activitystreams. If yes, parse for `name` within `content`. If a match is found, add a quote of that object (and optionally replace `name` in `content` with a link targeting `href`).

Reply:
Update on the FEP again: given that clients MUST specify `Accept: application/ld+json; profile="https://www.w3.org/ns/activitystreams"` it might make sense to use this `mediaType` instead of `application/activity+json`, since it is possible that an ActivityPub server might exist where `application/activity+json` is not supported (it is only a SHOULD and not a MUST).